src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -194,121 +194,166 @@ public static String composeRpcMethodHeaderSampleCode(
       List<MethodArgument> arguments,
       Map<String, ResourceName> resourceNames,
       Map<String, Message> messageTypes) {
-    // TODO(summerji): Add other types RPC methods' sample code.
+    VariableExpr clientVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder()
+                .setName(JavaStyle.toLowerCamelCase(clientType.reference().name()))
+                .setType(clientType)
+                .build());
+
+    List<Expr> bodyExprs = new ArrayList<>();
+
+    // Assign method's arguments variable with the default values.
+    List<VariableExpr> rpcMethodArgVarExprs = createRpcMethodArgumentVariableExprs(arguments);
+    List<Expr> rpcMethodArgDefaultValueExprs =
+        createRpcMethodArgumentDefaultValueExprs(arguments, resourceNames);
+    List<Expr> rpcMethodArgAssignmentExprs =
+        createAssignmentsForVarExprsWithValueExprs(
+            rpcMethodArgVarExprs, rpcMethodArgDefaultValueExprs);
+    bodyExprs.addAll(rpcMethodArgAssignmentExprs);
+
+    List<Statement> bodyStatements = new ArrayList<>();
     if (method.isPaged()) {
-      // Find the repeated field.
-      Message methodOutputMessage = messageTypes.get(method.outputType().reference().simpleName());
-      Field repeatedPagedResultsField = methodOutputMessage.findAndUnwrapFirstRepeatedField();
-      Preconditions.checkNotNull(
-          repeatedPagedResultsField,
-          String.format(
-              "No repeated field found on message %s for method %s",
-              methodOutputMessage.name(), method.name()));
-
-      TypeNode repeatedResponseType = repeatedPagedResultsField.type();
-      return SampleCodeWriter.write(
-          composeUnaryPagedRpcMethodSampleCode(
-              method, clientType, repeatedResponseType, arguments, resourceNames));
-    }
-    if (method.hasLro()) {
-      return SampleCodeWriter.write(
-          composeUnaryLroRpcMethodSampleCode(method, clientType, arguments, resourceNames));
+      bodyStatements.addAll(
+          composeUnaryPagedRpcMethodSampleCodeBodyStatements(
+              method, clientVarExpr, rpcMethodArgVarExprs, bodyExprs, messageTypes));
+    } else if (method.hasLro()) {
+      bodyStatements.addAll(
+          composeUnaryLroRpcMethodSampleCodeBodyStatements(
+              method, clientVarExpr, rpcMethodArgVarExprs, bodyExprs));
+    } else {
```

Comment:
Do we need any extra handling for streaming methods?

Reply:
No need to handle streaming methods here in `composeRpcMethodHeaderSampleCode`.
The caller is `createMethodVariants` only pass in the non-stream method in: 
```
if (method.stream().equals(Stream.NONE)) {
        javaMethods.addAll(
            createMethodVariants(method, clientName, messageTypes, types, resourceNames));
        javaMethods.add(
            createMethodDefaultMethod(method, clientName, messageTypes, types, resourceNames));
      }
```